src/ctia/stores/es/store.clj

```diff
@@ -15,6 +20,23 @@
   [{:keys [conn]}]
   (es-conn/close conn))
 
+(defn all-pages-iteration
+  "Returns lazy iteration of consecutive calls to `query-fn` with pagination params.
+
+  Resulted data is a sequence of responses of a shape `{:data [,,,] :paging {:next {,,,}}}`"
+  ([query-fn] (all-pages-iteration query-fn {}))
+  ([query-fn {:keys [limit]
+              :or {limit default-limit}
+              :as params}]
+   (let [limit (if (> limit max-result-window)
```

Comment:
`max-result-window` is a default value which can be less or greater than the value in the index settings. Why not use the value that is in the settings?

Reply:
because this function intentionally doesn't know about settings. There is a place to provide a limit value explicitly via `params` and only if missing — fallback to the "very default" limit provided by ductile library.